src/main/java/org/spectra/cluster/model/consensus/GreedyConsensusSpectrum.java

```diff
@@ -303,10 +306,79 @@ private IBinarySpectrum generateConsensusSpectrum() {
         return adaptedPeaks;
     }
 
+    /**
+     * Adapt the peak intensities in allPeaksInCluster using the following formula:
+     * I = I * (0.95 + 0.05 * (1 + pi)^5) . This probability cames from the FrankEtAll manuscript.
+     *
+     *
+     * @param peaks This are all the peaks in the following consensus cluster.
+     * @param nSpectra Number of spectra.
+     * @return A clean array with {@link BinaryConsensusPeak}
+     */
+    protected  BinaryConsensusPeak[] adaptPeakWithNoiseFilterIntensities(BinaryConsensusPeak[] peaks, int nSpectra) {
+
+        BinaryConsensusPeak[] adaptedPeaks = new BinaryConsensusPeak[peaks.length];
+
+        double doubleSpectra = (double) nSpectra;
+
+        int maxMz = 0;
+
+        for (int i = 0; i < peaks.length; i++) {
+            double peakProbability = (double) peaks[i].getCount() / doubleSpectra;
+            int newIntensity = (int) Math.round((double) peaks[i].getIntensity() * (0.95 + 0.05 * Math.pow(1 + peakProbability, 5)));
+
+            adaptedPeaks[i] = new BinaryConsensusPeak(peaks[i].getMz(), newIntensity, peaks[i].getCount());
+            if(peaks[i].getMz() > maxMz)
+                maxMz = peaks[i].getMz();
+        }
+
+        if (adaptedPeaks.length < minPeaksToKeep || adaptedPeaks.length < 1) {
+            return adaptedPeaks;
+        }
+
+        int peakIndex = 0;
+
+        /**
+         * The maxiumn number of peaks to keep will be, the number of intervals * number of peaks per interval.
+         */
+        List<BinaryConsensusPeak> peaksToKeep = new ArrayList<>((maxMz/windowSize) * peaksPerWindowToKeep);
+
+        // Keep top N peaks per W m/z
+        for (int windowStart = 0; windowStart <= maxMz && peakIndex < adaptedPeaks.length; windowStart += windowSize) {
+            List<BinaryConsensusPeak> windowPeaks = new ArrayList<>((maxMz/windowSize) * peaksPerWindowToKeep);
```

Comment:
Why do you allocate the max number of peaks again? It's fine but I'd just was wondering. Also, since that number is used quite often, putting it in a variable before the `... peaksToKepp = ` might be more efficient.